lib/algolia/webmock.rb

```diff
@@ -0,0 +1,102 @@
+begin
+  require 'webmock'
+rescue LoadError
+  puts 'WebMock was not found, please add "gem \'webmock\'" to your Gemfile.'
+  exit 1
+end
+
+module Algolia
+  class WebMock
+    def self.mock!
+      # list indexes
+      ::WebMock.stub_request(:get, %r{.*\.algolia(net\.com|\.net)/1/indexes}).to_return(body: '{ "items": [] }')
```

Comment:
Personally, I find this quite hard to read (because of the repeated `%r{.*\.algolia(net\.com|\.net)/1/indexes}`) mostly. WDYT of refactoring it into a `mock_endpoint` method, so that we can do this:

```ruby
mock_endpoint(:get, '/1/indexes', { body: '{ "items": [] }' })
```

WDYT? 🙂

Reply:
I think it's indeed hard to read but at the same time it's not really supposed to be read, just used as an helper to stub calls to the API. Moreover it might be more helpful to read the mock as-is than having it refactored. I would be in favor of keeping it like this

Reply:
I'm not sure I agree 😅  I think it should be readable, because it's code that we (and other contributors) have to maintain, even if we don't have to touch it often. If we add a new domain (e.g. `algolia.dev` or something similar) it means we have to refactor the whole class, instead of just a single line of code. I think refactoring this will make contributing easier in the future, so I would be in favor of changing it 🙂

Reply:
It's true that it's better for refactoring 👍 I'll make the changes